pkg/util/volume.go

```diff
@@ -103,3 +103,36 @@ func IndexPodByPVCFunc(pod *corev1.Pod) ([]string, error) {
 	}
 	return indexes, nil
 }
+
+// parseJSONSetting parses a JSON string into the provided target interface.
+func parseJSONSetting(setting string, target interface{}) error {
+	if setting == "" {
+		return fmt.Errorf("setting is empty")
+	}
+	return json.Unmarshal([]byte(setting), target)
+}
+
+func GetCSIOnlineExpandValidation(
+	provisioner string,
+	settingCache ctlharvesterv1.SettingCache,
+) (bool, error) {
+	if provisioner == "" {
+		return false, fmt.Errorf("provisioner is empty")
+	}
+
+	coevSetting, err := settingCache.Get(settings.CSIOnlineExpandValidationSettingName)
+	if err != nil {
+		return false, fmt.Errorf("failed to get %s setting: %w", settings.CSIOnlineExpandValidationSettingName, err)
+	}
+
+	coev := make(map[string]bool)
+	for _, data := range []string{coevSetting.Default, coevSetting.Value} {
```

Comment:
should we merge the maps? the default will always have `longhorn` csi expansion enabled. Unless user explicitly adds `longhorn` provider to the list of values it will be dropped during validation